duckduckgo/duckduckgo-privacy-extension
# Tighten ATB, add unit tests

**Reviewer:** @jdorweiler 
**CC:** @jkv 

## Description:
- Tightens up ATB pipeline steps
- Adds `dev=1` to all requests made from dev builds of the extension
- Adds the ability to handle errors from the load module
- ... and adds unit test coverage as for the ATB module as much as possible.

See inline comments for more info.

## Steps to test this PR:

Test the following two scenarios on Chrome, FF and Safari. Note, Safari may need to be put on DDH6 for us to test the entire flow.

**Case 1: no post-install success page (e.g. if the user installed via the Chrome web store)**

1. Install the extension with no DDG tabs open
2. The following requests should be made:
    - `atb.js`
    - `/exti?atb=v124-2`
3. Run a search. The following request should be made:
    - `atb.js?atb=v124-2&set_atb=v124-2`

**Case 2: post-install success page**

1. Open a browser tab for https://duckduckgo.com/?exti=2
2. Continue with all the steps in case 2.
3. The same URLs should be hit, only the variants should be added (e.g. `exti?atb=v124-2__` or `exti?atb=v124-2ab`

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
